python/taichi/profiler/profiler.py

```diff
@@ -0,0 +1,162 @@
+from taichi.core import ti_core as _ti_core
+from taichi.lang import impl
+
+import taichi as ti
+
+
+class StatisticalResult:
+    def __init__(self, name):
+        self.name = name
+        self.counter = 0
+        self.min_time = 0.0
+        self.max_time = 0.0
+        self.total_time = 0.0
+
+    def __lt__(self, other):
+        if (self.total_time < other.total_time):
+            return True
+        else:
+            return False
+
+    def insert_record(self, time):
+        if self.counter == 0:
+            self.min_time = time
+            self.max_time = time
+        self.counter += 1
+        self.total_time += time
+        self.min_time = min(self.min_time, time)
+        self.max_time = max(self.max_time, time)
+
+
+class Profiler:
+
+    profiling_modes = [True, False]
+
+    trace = 'trace'
+    count = 'count'
+    print_modes = [count, trace]
+
+    profiling_mode_ = False
+    total_time_ms_ = 0.0
+    traced_records_ = []
+    statistical_results_ = {}
+
+    def __init__(self):
+        _ti_core.trace(f'Profiler')
+
+    def set_kernel_profiler_mode(self, kernel_profiler=None):
+        if kernel_profiler is None:
+            self.profiling_mode_ = False
+        if kernel_profiler is False:
+            self.profiling_mode_ = False
+        elif kernel_profiler is True:
+            self.profiling_mode_ = True
+        else:
+            _ti_core.warn(
+                f'kernel_profiler mode error : {type(kernel_profiler)}')
```

Comment:
Why is this a ti_core.warning instead of a python warning? ;) 
(or even an error seems proper here) 
Maybe the code snippet below is equivalent and cleaner? 
```
def set_kernel_profiler_mode(self, kernel_profiler=False):
```